odl/tomo/backends/astra_setup.py

```diff
@@ -202,33 +202,40 @@ def astra_geom_to_vec(geometry):
         Numpy array of shape ``(number of angles, 12)``
     """
 
-    angles = geometry.angle_grid
-    num_angles = geometry.angle_grid.ntotal
+    angles = geometry.motion_grid
+    num_angles = geometry.motion_grid.ntotal
+
+    if isinstance(geometry, ConeBeamGeometry) and isinstance(geometry.detector, FlatDetector):
```

Comment:
It's two aspects, so two checks. We may want to have something like `is_curved` in the detector class in the long run.